crates/miden-testing/src/kernel_tests/tx/test_note.rs

```diff
@@ -54,7 +54,24 @@ use crate::{
 
 #[test]
 fn test_get_sender_no_sender() -> anyhow::Result<()> {
```

Comment:
I think I'd rewrite this test. We're manipulating kernel memory directly, which is error-prone and unnecessary. We should be able to call `miden::note::get_sender` from a transaction script, and it should fail with the same error. Finally, I'd call the test `note_get_sender_fails_from_tx_script` or something that better describes what it does.